src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import createFocusTrap from 'focus-trap';
+import { st, classes } from './Modal.st.css';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+import { ModalProps, ModalDefaultProps, ModalState } from './types';
+import { CloseIcon } from './Close';
+
+export class Modal extends React.Component<ModalProps, ModalState> {
+  static displayName = 'Modal';
+  static defaultProps: ModalDefaultProps = {
+    isOpen: false,
+    closeOnClickOutside: true,
+    minHeight: '150px',
+    maxHeight: '620px',
+    minWidth: 'auto',
+    maxWidth: '580px',
+    focusTrap: true,
+  };
+
+  el = null;
+  focusTrapInstance = null;
+
+  constructor(props: ModalProps) {
+    super(props);
+
+    this.el = document.createElement('div');
```

Comment:
This line won't work in SSR, because document doesn't exist.
So it should probably be moved to `componentDidMount`.
**However**, I don't think we need to create an extra `div`, maybe we can just use a ref?